astropy/io/ascii/html.py

```diff
@@ -354,6 +360,7 @@ def write(self, table):
             self.data.fill_values = [self.data.fill_values]
 
         self.data._set_fill_values(cols)
+        self.data._set_col_formats(cols)
```

Comment:
This needs a test.

Reply:
I've added an new check in `test_html.py` that checks things work as expected.

Reply:
> It doesn't work using the method already in BaseData as it says that the HTMLData does not have a cols attribute.

Ah I see, so I'd suggest that instead of making a new `_set_col_formats` method, a few lines up (after `self.data.header.cols = cols`) add:
```
self.data.cols = cols
```
Then call:
```
self.data._set_col_formats()
```
This matches the pattern in the `io.ascii` base classes as much as possible. I'm not sure why the data value processing was done differently in HTML to begin with (not using `self.str_vals()`), but that's a different story.

Reply:
Looks good, I'll make that change.

Reply:
I've checked this and it works, so the change has now been added.